hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinarySerializer.java

```diff
@@ -816,7 +802,8 @@ protected static BinaryId formatIndexId(HugeType type, Id indexLabel,
             id = HugeIndex.formatIndexHashId(type, indexLabel, fieldValues);
         }
         BytesBuffer buffer = BytesBuffer.allocate(1 + id.length());
-        return new BinaryId(buffer.writeId(id).bytes(), id);
+        byte[] idBytes = buffer.writeIndexId(id, type.isStringIndex()).bytes();
```

Comment:
please check idBytes length is 1 + id.length()

Reply:
still "1 + id.length()"